### PR TITLE
feat: add mempool chart and test in dev

### PR DESCRIPTION
**Pull request by openoms**

### CODE REVIEW

**Comment by openoms** [comment]
Just discussed with Wiz that to use the mempool backend for fee estimation only can disable the database (done in [162798a](https://github.com/GaloyMoney/charts/pull/3105/commits/162798a74767d0b4ed956c25d2332e0c2996349d)) and connect to the bitcoind RPC only, no Electrum backend is needed: `MEMPOOL_BACKEND = "none"`

File: charts/mempool/templates/pvc.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+apiVersion: v1
+metadata:
+  name: {{ template "mempool.fullname" . }}
+  annotations:
+    "helm.sh/resource-policy": keep
+  labels:
+    app: "{{ .Chart.Name }}"
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    release: "{{ .Release.Name }}"
+    heritage: "{{ .Release.Service }}"
```
Comment (krtk6160) [inline charts/mempool/templates/pvc.yaml:12]: I'm wondering if we should add these four labels to the other resources as well, or remove them from all. At the moment, it is inconsistent across our charts. For ex we use it in the [fulcrum pvc](https://github.com/galoymoney/charts/blob/ci--pin-gcp-provider-version/charts/fulcrum/templates/pvc.yaml#L9-L12) but not in its [statefulset](https://github.com/galoymoney/charts/blob/ci--pin-gcp-provider-version/charts/fulcrum/templates/statefulset.yaml#L6-L10).

I can see in the concourse chart it's being used uniformly for all resources, and we most likely got it from there.

File: ci/testflight/mempool/main.tf
Comment (krtk6160) [inline ci/testflight/mempool/main.tf:1]: Needs gcp provider version pinning like [here](https://github.com/GaloyMoney/charts/pull/3107)

File: ci/testflight/mempool/testflight-values.yml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+persistence:
+  enabled: false
+
+bitcoindRpcPort: 18443
+
+mariadb:
```
Comment (krtk6160) [inline ci/testflight/mempool/testflight-values.yml:6]: Leftover?

**Review by krtk6160 — CHANGES_REQUESTED** [review]
Have made some comments.

Also I don't think we need the rbac stuff - role, rolebinding, svc account creation. I don't see it being used anywhere in this chart.

File: ci/testflight/mempool/testflight-values.yml (hunk 2 above)
Comment (openoms) [reply ci/testflight/mempool/testflight-values.yml]: thanks addressed in [d049bc5](https://github.com/GaloyMoney/charts/pull/3105/commits/d049bc54c4bd46f83e99f6427d30bd76c9a4fbe5)

File: charts/mempool/templates/statefulset.yaml
Comment (krtk6160) [inline charts/mempool/templates/statefulset.yaml:1]: I think this is missing a volumeMount for the mempool container

**Review by krtk6160 — CHANGES_REQUESTED** [review]
One more comment

File: charts/mempool/templates/statefulset.yaml
Comment (openoms) [reply charts/mempool/templates/statefulset.yaml:1]: this showes that it doesn't even need a volume so removed all persistence

**Review by krtk6160 — APPROVED** [review]
LGTM

**Review by krtk6160 — APPROVED** [review]

**Pull request merged**